src/Zenova/Platform/Windows.cpp

```diff
@@ -87,6 +88,98 @@ namespace Zenova::PlatformImpl {
 			FreeLibraryAndExitThread(reinterpret_cast<HMODULE>(CleanupVariables), 0);
 		}
 	}
+
+	void* LoadModModuleAndResolveImports(const std::string& module)
+	{
+		void* hModule = LoadLibraryExA(module.c_str(), NULL, DONT_RESOLVE_DLL_REFERENCES);
+		ResolveModModuleImports(hModule, module);
+		return hModule;
+	}
+
+	typedef BOOL(APIENTRY* DllMainFunction)(HMODULE, DWORD, LPVOID);
+	void ResolveModModuleImports(void* hModule, const std::string& moduleName)
+	{
+		if (!hModule) {
+			Zenova_Log(Warning, "[{}] Could not resolve the imports for the module because it was nullptr.", __FUNCTION__);
+			Platform::DebugPause();
+			return;
+		}
+
+		uintptr_t hModuleAddress = reinterpret_cast<uintptr_t>(hModule);
+		PIMAGE_DOS_HEADER dosHeader = reinterpret_cast<PIMAGE_DOS_HEADER>(hModuleAddress);
+		PIMAGE_NT_HEADERS ntHeader = reinterpret_cast<PIMAGE_NT_HEADERS>(hModuleAddress + dosHeader->e_lfanew);
+
+		IMAGE_DATA_DIRECTORY importDir = ntHeader->OptionalHeader.DataDirectory[1];
+		PIMAGE_IMPORT_DESCRIPTOR imports = reinterpret_cast<PIMAGE_IMPORT_DESCRIPTOR>(hModuleAddress + importDir.VirtualAddress);
+		if (!imports) {
+			Zenova_Log(Warning, "[{}] Could not resolve the imports for the module because the imports data directory could not be found in the PE.", __FUNCTION__);
+			Platform::DebugPause();
+			return;
+		}
+
+		std::map<std::pair<uintptr_t, std::string>, ULONGLONG*> symbolTableAddresses;
+		while (imports->Characteristics != 0) {
```

Comment:
Can the while loops be swapped out with a for loop with an empty initialization?
```cpp
for (; cond; post)
```